src/js/utils/html-utils.js

```diff
@@ -6,41 +6,78 @@
+         */
+        function HtmlSnippet(htmlString) {
+            this.text = htmlString;
+        }
+        HtmlSnippet.prototype.valueOf = function() {
+            return this.text;
+        };
+        HtmlSnippet.prototype.toString = function() {
+            return this.text;
+        };
+
+        /**
+         * Helper function to create an HTML snippet from a string.
+         *
+         * @param {string} htmlString The string of HTML.
+         * @returns {HtmlSnippet} An HTML snippet that can be safely rendered.
          * @constructor
          */
         HTML = function(htmlString) {
-            var htmlSnippet = new String(htmlString); // jshint ignore:line
-            htmlSnippet.isHtmlSnippet = true;
-            return htmlSnippet;
+            return new HtmlSnippet(htmlString);
         };
 
         /**
-         * Returns an HTML string that is appropriately escaped.
+         * Ensures that the specified HTML is encoded as an HtmlSnippet.
          *
-         * Note: if the string provided is flagged as being an HTML snippet
+         * If the HTML is provided as a string, then it is HTML-escaped and
+         * returned as an HtmlSnippet. If the parameter is an HTML snippet
          * then it will be returned directly so as not to double escape it.
          *
-         * @param {string} text A string that is either plan text, or is
-         * flagged as being an HTML snippet.
-         * @returns {string} A safely HTML-escaped string.
+         * @param {(string|HtmlSnippet)} html HTML provided either as
+         * a string or as an HtmlSnippet.
+         * @returns {HtmlSnippet} A safely escaped HTML snippet.
          */
-        escape = function(text) {
-            if (text.isHtmlSnippet) {
-                return text.toString();
+        ensureHtml = function(html) {
+            if (html instanceof HtmlSnippet) {
+                return html;
             } else {
-                return _.escape(text);
+                return HTML(_.escape(html));
             }
         };
 
         /**
+         * A wrapper for $ that returns a new JQuery element for the specified HTML.
+         *
+         * Note that the HTML must have a single root element which will be the
+         * one returned. An error will be thrown if the HTML cannot be interpreted
+         * as a single root element.
+         *
+         * @param {HtmlSnippet} html The desired HTML provided as an HtmlSnippet.
+         * @returns {element} A new JQuery element.
+         */
+        element = function(html) {
+            if (!(html instanceof HtmlSnippet)) {
+                throw 'HtmlUtils.element received HTML that was not an HtmlSnippet: ' + html;
+            }
+            return $(ensureHtml(html).toString());
+        };
+
+        /**
          * Returns an HTML snippet by interpolating the provided parameters.
          *
          * The HTML text is provided as a tokenized format string where parameters
```

Comment:
The text is provided...

It's probably plain text that is being provided, not HTML.